runtime/tests/checker/interface_test.go

```diff
@@ -2689,3 +2689,1069 @@ func TestCheckBadStructInterface(t *testing.T) {
+		require.ErrorAs(t, errs[0], &conformanceError)
+		assert.Equal(t, "B", conformanceError.InterfaceType.QualifiedIdentifier())
+		assert.Equal(t, "A", conformanceError.NestedInterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[1], &conformanceError)
+		assert.Equal(t, "Q", conformanceError.InterfaceType.QualifiedIdentifier())
+		assert.Equal(t, "P", conformanceError.NestedInterfaceType.QualifiedIdentifier())
+	})
+
+	t.Run("same conformance via different paths", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface P: A {}
+
+            struct interface Q: A {}
+
+            struct interface X: P, Q {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("same conformance via different paths for struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface P: A {}
+
+            struct interface Q: A {}
+
+            struct X: P, Q {}
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("cyclic conformance", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface Foo: Baz {
+                let x: Int
+
+                fun test(): Int
+            }
+
+            struct interface Bar: Foo {}
+
+            struct interface Baz: Bar {}
+        `)
+
+		errs := RequireCheckerErrors(t, err, 3)
+
+		conformanceError := sema.CyclicConformanceError{}
+		require.ErrorAs(t, errs[0], &conformanceError)
+		assert.Equal(t, "Foo", conformanceError.InterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[1], &conformanceError)
+		assert.Equal(t, "Bar", conformanceError.InterfaceType.QualifiedIdentifier())
+
+		require.ErrorAs(t, errs[2], &conformanceError)
+		assert.Equal(t, "Baz", conformanceError.InterfaceType.QualifiedIdentifier())
+	})
+}
+
+func TestCheckInterfaceDefaultMethodsInheritance(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("default impl in super", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+
+            struct interface B: A {
+                pub fun hello()
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("default impl in child", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {
+                pub fun hello()
+            }
+
+            struct interface B: A {
+                pub fun hello() {
+                    var a = 1
+                }
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("default impl in both", func(t *testing.T) {
```

Comment:
Can we get a test for `A` with a default impl, `B` inheriting from `A` with no conflict, and then `C` inheriting from `B` that has another default impl to conflict with `A`?

Reply:
Added in c535fb201ed7837b5b9b65bc852a796b801fc8da